x-pack/plugins/security_solution/public/common/containers/matrix_histogram/use_matrix_histogram_async.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { useAsync, withOptionalSignal } from '../../../shared_imports';
+import { DataPublicPluginStart } from '../../../../../../../src/plugins/data/public';
+import { MatrixHistogramQueryProps } from '../../components/matrix_histogram/types';
+import { createFilter } from '../../../common/containers/helpers';
+import {
+  MatrixHistogramQuery,
+  MatrixHistogramRequestOptions,
+  MatrixHistogramStrategyResponse,
+  MatrixHistogramData,
+} from '../../../../common/search_strategy/security_solution';
+import { isErrorResponse, isCompleteResponse } from '../../../../../../../src/plugins/data/common';
+import { getInspectResponse } from '../../../helpers';
+import { InspectResponse } from '../../../types';
+
+export interface GetMatrixHistogramProps extends Omit<MatrixHistogramQueryProps, 'errorMessage'> {
+  data: DataPublicPluginStart;
+  signal: AbortSignal;
+}
+
+export interface GetMatrixHistogramReturnArgs {
+  data: MatrixHistogramData[];
+  inspect: InspectResponse;
+  totalCount: number;
+}
+
+export const getMatrixHistorgram = async ({
```

Comment:
Mispelling suggestion fix
```suggestion
export const getMatrixHistogram = async ({
```

Reply:
The "Historgram" is a new really cool graph type 😉  Yes, thanks!

Reply:
Fixed! https://github.com/elastic/kibana/pull/78985/commits/ee6e151500bb528d95407df13a5239d86e1b0055